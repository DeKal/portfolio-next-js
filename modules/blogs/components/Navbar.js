import React, { Fragment } from 'react'
import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import { Translate } from 'react-localize-redux'
import NavLogo from '~/modules/blogs/components/navbar/NavLogo'
import NavItem from '~/modules/blogs/components/navbar/NavItem'
import Search from '~/modules/blogs/components/navbar/Search'
import { PAGE_LIST } from '~/modules/blogs/consts/pages'
import { REDIRECT_LINK, LOGO_IMAGE } from '~/modules/blogs/consts/logo'

const NavBar = () => (
  <Fragment>
    <Navbar bg="light" variant="light">
      <Container>
        <NavLogo redirectLink={REDIRECT_LINK} image={LOGO_IMAGE} />
        <Nav className="mr-auto">
          {PAGE_LIST.map(page => (
            <NavItem key={page.name} href={page.url}>
              <Translate id={page.name} />
            </NavItem>
          ))}
        </Nav>
        <Search />
      </Container>
    </Navbar>
  </Fragment>
)

export default NavBar
