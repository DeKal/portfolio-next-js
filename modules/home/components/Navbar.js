import React, { useState } from 'react'

import styled from 'styled-components/macro'
import NavHeader from '~/modules/home/components/nav/NavHeader'
import NavContent from '~/modules/home/components/nav/NavContent'
import { INIT_NAV_CONTENT_SHOW } from '~/modules/home/consts/initState'

const Navbar = () => {
  const [isShowNavContent, switchNav] = useState(INIT_NAV_CONTENT_SHOW)

  return (
    <MainHeader>
      <NavHeader isShowNavContent={isShowNavContent} switchNav={switchNav} />
      <NavContent isShowNavContent={isShowNavContent} />
    </MainHeader>
  )
}

const MainHeader = styled.header`
  position: fixed;
  z-index: 100;
  width: 100vw;
  height: ${props => props.theme.navHeightWhenScrollingOnMobile};
`

export default Navbar
