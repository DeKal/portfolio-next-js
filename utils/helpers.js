import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
