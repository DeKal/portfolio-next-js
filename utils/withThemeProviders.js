import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
