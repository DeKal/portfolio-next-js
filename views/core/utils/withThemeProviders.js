import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
