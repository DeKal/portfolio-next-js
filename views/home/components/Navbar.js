import React, { useState } from 'react'

import styled from 'styled-components/macro'
import NavHeader from '~/views/home/components/nav/NavHeader'
import NavContent from '~/views/home/components/nav/NavContent'
import { INIT_NAV_CONTENT_SHOW } from '~/views/home/consts/initState'

const Navbar = () => {
  const [isShowNavContent, switchNav] = useState(INIT_NAV_CONTENT_SHOW)

  return (
    <MainHeader>
      <NavHeader isShowNavContent={isShowNavContent} switchNav={switchNav} />
      <NavContent isShowNavContent={isShowNavContent} />
    </MainHeader>
  )
}

const MainHeader = styled.header`
  position: fixed;
  z-index: 100;
  width: 100vw;
  height: ${props => props.theme.navHeightWhenScrollingOnMobile};
`

export default Navbar
