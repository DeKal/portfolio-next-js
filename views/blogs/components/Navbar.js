import React, { Fragment } from 'react'
import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import NavLogo from '~/views/blogs/components/navbar/NavLogo'
import NavItem from '~/views/blogs/components/navbar/NavItem'
import Search from '~/views/blogs/components/navbar/Search'

const NavBar = () => (
  <Fragment>
    <Navbar bg="light" variant="light">
      <Container>
        <NavLogo redirectLink="/blogs" image="/images/logo.png" />
        <Nav className="mr-auto">
          <NavItem href="/">Home</NavItem>
          <NavItem href="/blogs">Blogs</NavItem>
        </Nav>
        <Search />
      </Container>
    </Navbar>
  </Fragment>
)

export default NavBar
