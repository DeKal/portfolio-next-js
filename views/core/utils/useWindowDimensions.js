import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
