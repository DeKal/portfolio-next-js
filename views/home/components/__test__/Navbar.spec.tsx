import * as React from 'react'
import { mountWithTheme } from '~/utils/withThemeProviders'
import Navbar from '~/views/home/components/Navbar'

describe('Navbar', () => {

  it('should render without throwing an error', function () {
    const wrap = mountWithTheme(<Navbar />)
    expect(wrap).toMatchSnapshot()
  })

})