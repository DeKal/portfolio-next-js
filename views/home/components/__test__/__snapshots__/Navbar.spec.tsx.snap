// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render without throwing an error 1`] = `
<Navbar>
  <Navbar__MainHeader>
    <header
      className="Navbar__MainHeader-sc-24flfj-0 cPsmOv"
    >
      <NavHeader
        isShowNavContent={false}
        switchNav={[Function]}
      >
        <NavHeader__NavContainer>
          <div
            className="NavHeader__NavContainer-ustrcb-0 ektEKS"
          >
            <MobileContainer>
              <Container
                className="MobileContainer-sc-1lm99xs-0 grSHFn"
                fluid={false}
              >
                <div
                  className="MobileContainer-sc-1lm99xs-0 grSHFn container"
                >
                  <ForwardRef
                    fluid="true"
                    noGutters={false}
                  >
                    <div
                      className="row"
                      fluid="true"
                    >
                      <Col
                        xs={6}
                      >
                        <div
                          className="col-6"
                        >
                          <NavLang>
                            <NavLang__NotSelectableDiv>
                              <nav
                                className="NavLang__NotSelectableDiv-aysloj-0 eeBxyb"
                              >
                                <ul>
                                  <NavLang__NavLinkItem>
                                    <li
                                      className="NavLang__NavLinkItem-aysloj-1 fnEXEp"
                                    >
                                      <LocalizedComponent
                                        lang="en"
                                        selectedLang="en"
                                        setSelectedLang={[Function]}
                                      >
                                        <NavLink
                                          addTranslation={[Function]}
                                          addTranslationForLanguage={[Function]}
                                          ignoreTranslateChildren={false}
                                          initialize={[Function]}
                                          lang="en"
                                          languages={Array []}
                                          renderToStaticMarkup={false}
                                          selectedLang="en"
                                          setActiveLanguage={[Function]}
                                          setSelectedLang={[Function]}
                                          translate={[Function]}
                                        >
                                          <NavLink__Container
                                            isSelected={true}
                                            onClick={[Function]}
                                          >
                                            <a
                                              className="NavLink__Container-sc-14bn6cd-0 bZIfvx"
                                              onClick={[Function]}
                                            >
                                              <Translate
                                                id="english"
                                              >
                                                <WrappedTranslate
                                                  context={
                                                    Object {
                                                      "activeLanguage": undefined,
                                                      "addTranslation": [Function],
                                                      "addTranslationForLanguage": [Function],
                                                      "defaultLanguage": undefined,
                                                      "ignoreTranslateChildren": false,
                                                      "initialize": [Function],
                                                      "languages": Array [],
                                                      "renderToStaticMarkup": false,
                                                      "setActiveLanguage": [Function],
                                                      "translate": [Function],
                                                    }
                                                  }
                                                  id="english"
                                                >
                                                  Missing translationId: english for language: \${ languageCode }
                                                </WrappedTranslate>
                                              </Translate>
                                            </a>
                                          </NavLink__Container>
                                        </NavLink>
                                      </LocalizedComponent>
                                      <NavLang__InlineBlock>
                                        <div
                                          className="NavLang__InlineBlock-aysloj-2 dCEQtc"
                                        >
                                          /
                                        </div>
                                      </NavLang__InlineBlock>
                                      <LocalizedComponent
                                        lang="vn"
                                        selectedLang="en"
                                        setSelectedLang={[Function]}
                                      >
                                        <NavLink
                                          addTranslation={[Function]}
                                          addTranslationForLanguage={[Function]}
                                          ignoreTranslateChildren={false}
                                          initialize={[Function]}
                                          lang="vn"
                                          languages={Array []}
                                          renderToStaticMarkup={false}
                                          selectedLang="en"
                                          setActiveLanguage={[Function]}
                                          setSelectedLang={[Function]}
                                          translate={[Function]}
                                        >
                                          <NavLink__Container
                                            isSelected={false}
                                            onClick={[Function]}
                                          >
                                            <a
                                              className="NavLink__Container-sc-14bn6cd-0 gBWpuU"
                                              onClick={[Function]}
                                            >
                                              <Translate
                                                id="vietnamese"
                                              >
                                                <WrappedTranslate
                                                  context={
                                                    Object {
                                                      "activeLanguage": undefined,
                                                      "addTranslation": [Function],
                                                      "addTranslationForLanguage": [Function],
                                                      "defaultLanguage": undefined,
                                                      "ignoreTranslateChildren": false,
                                                      "initialize": [Function],
                                                      "languages": Array [],
                                                      "renderToStaticMarkup": false,
                                                      "setActiveLanguage": [Function],
                                                      "translate": [Function],
                                                    }
                                                  }
                                                  id="vietnamese"
                                                >
                                                  Missing translationId: vietnamese for language: \${ languageCode }
                                                </WrappedTranslate>
                                              </Translate>
                                            </a>
                                          </NavLink__Container>
                                        </NavLink>
                                      </LocalizedComponent>
                                    </li>
                                  </NavLang__NavLinkItem>
                                </ul>
                              </nav>
                            </NavLang__NotSelectableDiv>
                          </NavLang>
                        </div>
                      </Col>
                      <Col
                        xs={6}
                      >
                        <div
                          className="col-6"
                        >
                          <NavHeader__NavControl
                            data-test-id="switch-nav-content"
                            onClick={[Function]}
                          >
                            <div
                              className="NavHeader__NavControl-ustrcb-1 hNFHhI"
                              data-test-id="switch-nav-content"
                              onClick={[Function]}
                            >
                              <NavHeader__HamburgerIcon>
                                <GiHamburgerMenu
                                  className="NavHeader__HamburgerIcon-ustrcb-2 hVcfUK"
                                >
                                  <IconBase
                                    attr={
                                      Object {
                                        "viewBox": "0 0 512 512",
                                      }
                                    }
                                    className="NavHeader__HamburgerIcon-ustrcb-2 hVcfUK"
                                  >
                                    <svg
                                      className="NavHeader__HamburgerIcon-ustrcb-2 hVcfUK"
                                      fill="currentColor"
                                      height="1em"
                                      stroke="currentColor"
                                      strokeWidth="0"
                                      style={
                                        Object {
                                          "color": undefined,
                                        }
                                      }
                                      viewBox="0 0 512 512"
                                      width="1em"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
                                        key="0"
                                      />
                                    </svg>
                                  </IconBase>
                                </GiHamburgerMenu>
                              </NavHeader__HamburgerIcon>
                               
                              <Translate
                                id="Menu"
                              >
                                <WrappedTranslate
                                  context={
                                    Object {
                                      "activeLanguage": undefined,
                                      "addTranslation": [Function],
                                      "addTranslationForLanguage": [Function],
                                      "defaultLanguage": undefined,
                                      "ignoreTranslateChildren": false,
                                      "initialize": [Function],
                                      "languages": Array [],
                                      "renderToStaticMarkup": false,
                                      "setActiveLanguage": [Function],
                                      "translate": [Function],
                                    }
                                  }
                                  id="Menu"
                                >
                                  Missing translationId: Menu for language: \${ languageCode }
                                </WrappedTranslate>
                              </Translate>
                            </div>
                          </NavHeader__NavControl>
                        </div>
                      </Col>
                    </div>
                  </ForwardRef>
                </div>
              </Container>
            </MobileContainer>
          </div>
        </NavHeader__NavContainer>
      </NavHeader>
      <NavContent
        isShowNavContent={false}
      >
        <NavContent__NavContainer
          isShowNavContent={false}
        >
          <div
            className="NavContent__NavContainer-sc-1fjhoh9-0 egWEWD"
          >
            <MobileContainer>
              <Container
                className="MobileContainer-sc-1lm99xs-0 grSHFn"
                fluid={false}
              >
                <div
                  className="MobileContainer-sc-1lm99xs-0 grSHFn container"
                >
                  <ForwardRef
                    noGutters={false}
                  >
                    <div
                      className="row"
                    >
                      <Col
                        lg={2}
                      >
                        <div
                          className="col-lg-2"
                        >
                          <NavLogo>
                            <NavLogo__Logo
                              href="https://github.com/DeKal/portfolio-next-js"
                            >
                              <a
                                className="NavLogo__Logo-sc-94vjc7-0 invqvT"
                                href="https://github.com/DeKal/portfolio-next-js"
                              >
                                <NavLogo__LogoText>
                                  <span
                                    className="NavLogo__LogoText-sc-94vjc7-1 iMA-dJr"
                                  >
                                    JD
                                  </span>
                                </NavLogo__LogoText>
                              </a>
                            </NavLogo__Logo>
                          </NavLogo>
                        </div>
                      </Col>
                      <Col
                        fluid="true"
                      >
                        <div
                          className="col"
                          fluid="true"
                        >
                          <NavContent__NavMenu>
                            <nav
                              className="NavContent__NavMenu-sc-1fjhoh9-1 hJvlsz"
                            >
                              <ul>
                                <NavContent__NavLi
                                  key="Home"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "Home",
                                          "name": "Home",
                                          "url": "/#Home",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/#Home"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/#Home"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="Home"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Home"
                                            >
                                              Missing translationId: Home for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                                <NavContent__NavLi
                                  key="About"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "About",
                                          "name": "About",
                                          "url": "/#About",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/#About"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/#About"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="About"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="About"
                                            >
                                              Missing translationId: About for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                                <NavContent__NavLi
                                  key="Experiences"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "Experiences",
                                          "name": "Experiences",
                                          "url": "/#Experiences",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/#Experiences"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/#Experiences"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="Experiences"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Experiences"
                                            >
                                              Missing translationId: Experiences for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                                <NavContent__NavLi
                                  key="Projects"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "Projects",
                                          "name": "Projects",
                                          "url": "/#Projects",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/#Projects"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/#Projects"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="Projects"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Projects"
                                            >
                                              Missing translationId: Projects for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                                <NavContent__NavLi
                                  key="Certifications"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "Certifications",
                                          "name": "Certifications",
                                          "url": "/#Certifications",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/#Certifications"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/#Certifications"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="Certifications"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Certifications"
                                            >
                                              Missing translationId: Certifications for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                                <NavContent__NavLi
                                  key="Contacts"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "Contacts",
                                          "name": "Contacts",
                                          "url": "/#Contacts",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/#Contacts"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/#Contacts"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="Contacts"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Contacts"
                                            >
                                              Missing translationId: Contacts for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                                <NavContent__NavLi
                                  key="Blogs"
                                >
                                  <li
                                    className="NavContent__NavLi-sc-1fjhoh9-2 uiPJU"
                                  >
                                    <NavItem
                                      isSelected={false}
                                      page={
                                        Object {
                                          "cmpName": "blogs",
                                          "name": "Blogs",
                                          "url": "/blogs",
                                        }
                                      }
                                      setSelectedPage={[Function]}
                                    >
                                      <NavItem__Container
                                        data-test-id="nav-link"
                                        href="/blogs"
                                        onClick={[Function]}
                                      >
                                        <a
                                          className="NavItem__Container-sc-1wmlvlx-0 ygPuK"
                                          data-test-id="nav-link"
                                          href="/blogs"
                                          onClick={[Function]}
                                        >
                                          <Translate
                                            id="Blogs"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Blogs"
                                            >
                                              Missing translationId: Blogs for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </a>
                                      </NavItem__Container>
                                    </NavItem>
                                  </li>
                                </NavContent__NavLi>
                              </ul>
                            </nav>
                          </NavContent__NavMenu>
                        </div>
                      </Col>
                    </div>
                  </ForwardRef>
                </div>
              </Container>
            </MobileContainer>
          </div>
        </NavContent__NavContainer>
      </NavContent>
    </header>
  </Navbar__MainHeader>
</Navbar>
`;
